src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/BookingSearchTest.kt

```diff
@@ -406,4 +541,102 @@ class BookingSearchTest : IntegrationTestBase() {
       },
     )
   }
+
+  private fun create10TestTemporaryAccommodationBookings(
+    userEntity: UserEntity,
+    offenderDetails: OffenderDetailSummary,
+  ): MutableList<BookingEntity> {
+    return createTestTemporaryAccommodationBookings(userEntity, offenderDetails, 5, 2)
+  }
+
+  private fun create15TestTemporaryAccommodationBookings(
+    userEntity: UserEntity,
+    offenderDetails: OffenderDetailSummary,
+  ): MutableList<BookingEntity> {
+    return createTestTemporaryAccommodationBookings(userEntity, offenderDetails, 5, 3)
+  }
+
+  private fun createTestTemporaryAccommodationBookings(
+    userEntity: UserEntity,
+    offenderDetails: OffenderDetailSummary,
+    numberOfPremises: Int,
+    numberOfBedsInEachPremises: Int,
+  ): MutableList<BookingEntity> {
+    val allPremises = temporaryAccommodationPremisesEntityFactory.produceAndPersistMultiple(numberOfPremises) {
+      withProbationRegion(userEntity.probationRegion)
+      withYieldedLocalAuthorityArea {
+        localAuthorityEntityFactory.produceAndPersist()
+      }
+    }
+
+    val allBeds = mutableListOf<BedEntity>()
+    allPremises.forEach { premises ->
+      val rooms = roomEntityFactory.produceAndPersistMultiple(numberOfBedsInEachPremises) {
+        withPremises(premises)
+      }
+
+      rooms.forEach { room ->
+        val bed = bedEntityFactory.produceAndPersist {
+          withRoom(room)
+        }
+
+        allBeds += bed
+      }
+    }
+
+    val allBookings = mutableListOf<BookingEntity>()
+    allBeds.forEachIndexed { index, bed ->
+      val booking = bookingEntityFactory.produceAndPersist {
+        withPremises(bed.room.premises)
+        withCrn(offenderDetails.otherIds.crn)
+        withBed(bed)
+        withServiceName(ServiceName.temporaryAccommodation)
+        withArrivalDate(LocalDate.now().minusDays((60 - index).toLong()))
+        withDepartureDate(LocalDate.now().minusDays((30 - index).toLong()))
+      }
+
+      allBookings += booking
+    }
+    return allBookings
+  }
+
+  private fun createApprovedBookingEntities(
```

Comment:
This should probably be called `createApprovedPremisesBookingEntities` as Approved Premises is both the name of the CAS1 service and the name for an individual premises within the service.

Reply:
done